# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

In [11]:
#Pregunta 1: el orden de ambas matrices es de 3x4 3 filas y 4 columnas

import numpy as np
A = np.array([[2.6, 4.8, 1.8, 0.9],[3.2, 4.4, 2.5, 2.8],[2.4, 3.6, 3.8, 2.5]])

B = np.array([[3.6, 2.5, 3.0, 2.5],[4.5, 5.0, 3.5, 3.8],[2.9, 3.0, 4.6, 4.0]])
#Pregunta2
S = A + B
print(f'S= {S}.')
#Pregunta 3
D = B-A
print(f'D = {D}.')
#4. Indique el valor e interprete los elementos de "s_24" y "d_24"
s_24 =S[2-1][4-1] #Fila 2: producto2 ; Columna 4: region 4. es la 4-1 ya que comienza en 0 python
d_24 = D [2-1][4-1]
print(d_24) #d_24=1 
print(f'El total de ventas, del producto 2 en la región 4 fue de {s_24}'\
    'millones de dólares, entre ambos años')
print(f'La diferenia entre la veta entre segundo y primer año del producto 2 en la region 4'\
    'fue de {d_24} millones de dolares')
#5.
print('No, no es posible determinar la matriz T = A + B + E ya que las matrices no tienen el mismo ordem')


S= [[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]].
D = [[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]].
1.0
El total de ventas, del producto 2 en la región 4 fue de 6.6millones de dólares, entre ambos años
La diferenia entre la veta entre segundo y primer año del producto 2 en la region 4fue de {d_24} millones de dolares
No, no es posible determinar la matriz T = A + B + E ya que las matrices no tienen el mismo ordem


# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

In [ ]:
#El orden de cada matriz es de: 5x3 5 filas y 3 colmnas
#Elemeto 
import numpy as np
J = np.array([[100,200,150],[120,210,170],[112,200,131],[124,310,170],[136,156,140]])
C = np.array([[110,190,160],[130,220,180],[146,190,141],[143,312,163],[137,116,156]])
#p2 j21 y c12 filas: modelos / columnas: componentes
j21 = J[2-1][1-1]#120
print(j21)
c12=C[1-1][2-1]
print(f'El costo de produccion del 1er componente del modelo 2 es {j21} mil $US en la sucursal de Japón')
print(f'El costo de produccion del 2do componente  del modelo 1 es {c12} mil $US en la sucursal de China')


120
El costo de produccion del 1er componente del modelo 2 es 120 mil $US en la sucursal de Japón
El costo de produccion del 2do componente  del modelo 1 es 190 mil $US en la sucursal de China


In [1]:
import numpy as np
J = np.array([[100,200,150],[120,210,170],[112,200,131],[124,310,170],[136,156,140]])
C = np.array([[110,190,160],[130,220,180],[146,190,141],[143,312,163],[137,116,156]])

suma_JM = np.sum(J, axis=1)
print(suma_JM)
suma_CM = np.sum(C, axis=1)
print(suma_CM)

[450 500 443 604 432]
[460 530 477 618 409]


In [ ]:
#4. un trabajador indica que, en la sucursal japonesa, el costo de produccion del "componente" 3, modelo 4 de computador, es más alto
#que en la sucursal china ¿Tiene razon?
#Justifique apropiadamante
import numpy as np
J = np.array([[100,200,150],[120,210,170],[112,200,131],[124,310,170],[136,156,140]])
C = np.array([[110,190,160],[130,220,180],[146,190,141],[143,312,163],[137,116,156]])
j43=J[4-1,3-1] #es lo mismo que [4-1][3-1]
c43=C[4-1,3-1]
print(j43)
print(c43)
print('El trabajador tiene raazon porque en la planta de japon tiene costo de 7mil dolares más')

170
163
El trabajador tiene raazon porque en la planta de japon tiene costo de 7mil dolares más


# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

In [ ]:
import numpy as np
C = np.array([[120,200,155,125,230],[155,130,110,160,180],[140,205,150,125,155],[130,120,230,225,125],[135,70,190,175,95],[110,265,180,230,180],[185,85,120,135,150]])
E = C-(C*0.2)
print(E)
#2
print('La exprecion que calcula la matriz tij =  cij + eij')
#3
T=C+E
print(T)
#4
c_34 = C[3-1][4-1]
t_72 = T[7-1][7-1]
print('c34 y t72 filas: unidades del producto / columnas: plantas de Beta')

[[ 96. 160. 124. 100. 184.]
 [124. 104.  88. 128. 144.]
 [112. 164. 120. 100. 124.]
 [104.  96. 184. 180. 100.]
 [108.  56. 152. 140.  76.]
 [ 88. 212. 144. 184. 144.]
 [148.  68.  96. 108. 120.]]
La exprecion que calcula la matriz tij =  cij + eij
[[216. 360. 279. 225. 414.]
 [279. 234. 198. 288. 324.]
 [252. 369. 270. 225. 279.]
 [234. 216. 414. 405. 225.]
 [243. 126. 342. 315. 171.]
 [198. 477. 324. 414. 324.]
 [333. 153. 216. 243. 270.]]
c34 y t72 filas: unidades del producto / columnas: plantas de Beta


# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.
